Docs-Guides - All Guides Converted to Sphinx

tonybove-apple opened the pull request:
This PR is for the GitHub-hosted public version of `coremltools`.

It adds a folder, `docs-guides`, and files converted from the public [Guides](https://coremltools.readme.io/docs) in the ReadMe.io system. This folder is not visible unless if you have the link to it, and the link is not yet public.

This PR includes all guide files converted and edited in Markdown, with ReStructured Text (RST) section pages for navigation and TOC. 

Added to requirements (`reqs/docs.pip`):
`myst-parser`
`sphinx-copybutton`
`sphinx-book-theme`


Added to repo:
`docs-guides` folder


Branch:
docs-all-guides-converted-to-sphinx

---

**Preview**:

- [Using the book theme](https://tonybove-apple.github.io/coremltools/docs-guides)

tonybove-apple commented:
> * In the "Overview" section, I think we can fold the "migration workflow" under the New features section. I dont think so we need to highlight it too much now and give it its own top level section, since that page if for migrating from coremltools 3 to coremltools 4 which is several years old. This will also shorten the overall length of the "overview" section in the landing page

Done. 

> * I like that you renamed this page more generally , calling it [coremltools conversion API](https://tonybove-apple.github.io/coremltools/docs-guides/source/unified-conversion-api.html) instead of "unified conversion api". However, if we call it that, we should at least add a sentence in the beginning, saying that this section covers conversion of neural network models and for other classic models go [here](link to the "Other Converters" section)

Done 

> * I dont think we need the "API Github" section/link under the "Essentials" heading on the landing page. The Github link is always on the top. Also that link is to the GitHub repo with all the source code, not just to the "APIs", so calling it that is a bit odd as well. We dont need it, we can remove it.

Done

> * In the MLPrograms section, I think there are too many drop downs. The "Save ML Programs as Model Packages" and "Find the Model Type in a Model Package" are very short sections and like add ons, so like the readme guide they can be folded into the main page and dont need their own sub sections.

Once I had folded "Save ML Programs as Model Packages" and "Find the Model Type in a Model Package" back into "Convert Models to ML Programs" I decided to reorganize the sections into one "Converting Deep Learning Models" section, with a separate page for converting neural networks. I also added "Workflow" to the "Load and Convert", and combined source and target formats into one page.

aseemw commented:
- Did we remove [this section](https://coremltools.readme.io/docs/new-in-coremltools#saving-a-core-ml-model-package) or move it to a different location?

aseemw commented:
In this [section](https://tonybove-apple.github.io/coremltools/docs-guides/source/target-conversion-formats.html#format-differences), the table seems redundant as the information is repeated below it.

aseemw commented:
Other than the 2 comments above, LGTM!

tonybove-apple commented:
> Did you remove [this section](https://coremltools.readme.io/docs/new-in-coremltools#saving-a-core-ml-model-package) or move it to a different location?

Yes, moved it into Convert Models to ML Programs. I also changed all the links throughout to `convert-to-ml-program.html#save-ml-programs-as-model-packages`.

tonybove-apple commented:
> In this [section](https://tonybove-apple.github.io/coremltools/docs-guides/source/target-conversion-formats.html#format-differences), the table seems redundant as the information is repeated below it.

I kept it as a summary at the top, for quick viewing.

tonybove-apple commented:
Merging now. We can continue to make improvements before going live with this. Thanks!
